### PR TITLE
Adding MDFactory client

### PR DESCRIPTION
**Pull request by lucis**
- [x] Adds the Master Data Factory classes that generates ready-to-use Master Data clients for VTEX IO
- [x] Add the reference on README.md
- [x] Sets prettier's `lineWidth`'s as 80

### CODE REVIEW

File: src/clients/masterData/masterDataFactory.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  InstanceOptions,
+  IOContext,
+  IOResponse,
+  JanusClient,
+  MasterData,
+  parseAppId,
+} from '@vtex/api'
+
+export const toStringIfNot = <K>(field: ThisType<K>) =>
+  typeof field === 'string' ? field : field?.toString()
```
Comment (mcandeia) [inline src/clients/masterData/masterDataFactory.ts]: It's not necessary, try field?.toString() directly
```suggestion
export const toStringIfNot = <K>(field: ThisType<K>) =>
  field?.toString()
```

File: src/clients/masterData/masterDataFactory.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+const GLOBAL = ''
+/**
+ * This is necessary since masterdata does not accept special characters on entity name
+ * This function replaces `.` and `-` for `_`
+ * @param str dataEntityName
+ */
+const normalizeEntityName = (str: string) => str.replace(/(\.)|-|:/gi, '_')
+
+const versionDescriptor = (isPublish: boolean, workspace: string) =>
```
Comment (mcandeia) [inline src/clients/masterData/masterDataFactory.ts]:
```suggestion
const versionDescriptor = (isProduction: boolean, workspace: string) =>
```

File: src/clients/masterData/masterDataFactory.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  isPublish ? GLOBAL : `-${workspace}`
```
Comment (mcandeia) [inline src/clients/masterData/masterDataFactory.ts]:
```suggestion
  isProduction ? GLOBAL : `-${workspace}`
```

**Review by mcandeia — APPROVED** [review]

File: src/clients/masterData/masterDataFactory.ts (hunk 1 above)
Comment (lucis) [reply src/clients/masterData/masterDataFactory.ts]: I think it makes sense because of you Typescript, no?

**Review by fabianaferreira — APPROVED** [review]
Niiiiiice 🚀

**Pull request merged**